homeUi/src/test/kotlin/com/gravatar/app/homeUi/presentation/home/gravatar/GravatarViewModelTest.kt

```diff
@@ -647,10 +649,75 @@ class GravatarViewModelTest {
         }
     }
 
+    @Test
+    fun `onEvent OnDownloadAvatar should download avatar image`() = runTest {
+        // Given
+        val avatars = createAvatars()
+        coEvery { userRepository.getAvatars() } returns Result.success(avatars)
+        initViewModel()
+        advanceUntilIdle()
+
+        val avatarId = "1"
+        val avatarUrl = avatars.first { it.imageId == avatarId }.imageUrl
+        coEvery { imageDownloader.downloadImage(avatarUrl) } returns GravatarResult.Success(Unit)
+
+        // When
+        viewModel.onEvent(GravatarEvent.OnDownloadAvatar(avatarId))
+        advanceUntilIdle()
+
+        // Then
+        coVerify { imageDownloader.downloadImage(avatarUrl) }
+    }
+
+    @Test
+    fun `onEvent OnDownloadAvatar should handle download manager not available`() = runTest {
```

Comment:
I decided to leave the failed tests and a comment in the view model, not to forget to update them when the error handling is implemented.